infra/update.go

```diff
@@ -0,0 +1,44 @@
+package infra
+
+import (
+	"github.com/gravitational/robotest/lib/defaults"
+	"github.com/gravitational/robotest/lib/wait"
+	"github.com/gravitational/trace"
+	"golang.org/x/crypto/ssh"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// UploadUpdate started uploading new application update
+// on installer node
+func UploadUpdate(provisioner Provisioner, installer Node) (err error) {
+	var session *ssh.Session
+	err = wait.Retry(defaults.RetryDelay, defaults.RetryAttempts, func() error {
+		session, err = installer.Connect()
+		if err != nil {
+			log.Debug(trace.DebugReport(err))
+		}
+		return trace.Wrap(err)
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer func() {
+		if err == nil {
+			return
```

Comment:
Well, in case of wizard the session is not discard like here but closed later in https://github.com/gravitational/robotest/blob/master/infra/wizard.go#L223